core_impl/src/main/java/com/google/instrumentation/stats/StatsSerializer.java

```diff
@@ -88,18 +89,18 @@ static void serialize(StatsContextImpl context, OutputStream output) throws IOEx
 
   // Deserializes input to StatsContext based on the binary format standard.
   // The encoded tags are of the form: <version_id><encoded_tags>
-  static StatsContextImpl deserialize(InputStream input) throws IOException {
+  static StatsContextImpl deserialize(InputStream input) throws IOException, ParseException {
```

Comment:
A problem with this: If they want to serialize() and deserialize() with byte[], for serialization they need to pass in an array that's large enough, but then there will probably be trailing 0s, and trying to deserialize that array will throw an exception. How are we supposed to resolve that?